qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/commands/QuteSurroundWithCommandHandler.java

```diff
@@ -0,0 +1,163 @@
+import com.redhat.qute.ls.commons.LineIndentInfo;
+import com.redhat.qute.ls.commons.snippets.SnippetsBuilder;
+import com.redhat.qute.parser.template.Node;
+import com.redhat.qute.parser.template.Section;
+import com.redhat.qute.parser.template.Template;
+import com.redhat.qute.settings.SharedSettings;
+
+/**
+ * Qute Command "qute.refactor.surround.with" to support surround:
+ * 
+ * <ul>
+ * <li>Surround with Section (Wrap)</li>
+ * <li>Surround with Comments</li>
+ * </ul>
+ *
+ */
+public class QuteSurroundWithCommandHandler extends AbstractTemplateCommandHandler {
+
+	public static final String COMMAND_ID = "qute.refactor.surround.with";
+
+	public static class SurroundWithResponse {
+
+		private TextEdit start;
+
+		private TextEdit end;
+
+		public SurroundWithResponse() {
+		}
+
+		public SurroundWithResponse(TextEdit start, TextEdit end) {
+			this.start = start;
+			this.end = end;
+		}
+
+		public TextEdit getStart() {
+			return start;
+		}
+
+		public TextEdit getEnd() {
+			return end;
+		}
+	}
+
+	public static enum SurroundWithKind {
+		section, //
+		comments;
+
+		public static SurroundWithKind get(String kind) {
+			return valueOf(kind);
+		}
+	}
+
+	public QuteSurroundWithCommandHandler(QuteTemplateProvider templateProvider) {
+		super(templateProvider);
+	}
+
+	@Override
+	protected CompletableFuture<Object> executeCommand(Template template, ExecuteCommandParams params,
+			SharedSettings sharedSettings,
+			CancelChecker cancelChecker) throws Exception {
+		// Get parameters
+		Range selection = ArgumentsUtils.getArgAt(params, 1, Range.class);
+		SurroundWithKind kind = SurroundWithKind.get(ArgumentsUtils.getArgAt(params, 2, String.class));
+		boolean snippetsSupported = ArgumentsUtils.getArgAt(params, 3, Boolean.class);
+
+		// Surround process
+		boolean emptySelection = selection.getStart().equals(selection.getEnd());
+		StringBuilder startText = null;
+		StringBuilder endText = null;
+		Position startPos = selection.getStart();
+		Position endPos = selection.getEnd();
+
+		int offset = template.offsetAt(selection.getStart());
+		Node node = template.findNodeAt(offset);
+
+		// Adjust selection if needed
+		boolean adjusted = false;
+		if (emptySelection && node instanceof Section) {
+			Section section = (Section) node;
+			if (section.isInStartTagName(offset) || section.isInEndTagName(offset)) {
+				startPos = template.positionAt(section.getStart());
+				endPos = template
+						.positionAt(section.isEndTagClosed() ? section.getEndTagCloseOffset() + 1 : section.getEnd());
+				adjusted = true;
+			}
+		}
+
+		switch (kind) {
+			case comments:
+				startText = new StringBuilder("{!");
+				if (snippetsSupported && emptySelection) {
+					SnippetsBuilder.tabstops(1, startText);
+				}
+				endText = new StringBuilder("!}");
+				if (snippetsSupported) {
+					SnippetsBuilder.tabstops(0, endText);
+				}
+				break;
+			default:
+				List<String> sections = getSectionNames(node, offset, adjusted);
```

Comment:
Indeed it should be really nice to support snippet with section, but let's do that in an another PR, let's generate only for the moment:

 * {#
 * {/}

And https://github.com/redhat-developer/quarkus-ls/issues/802 should takes care of apply completion of section (without snippet)